BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -1748,56 +1648,52 @@ unsafe public override IProjectiveTransformation GetIntrinsics(string channelNam
                 intrinsics.ppy,
                 intrinsics.coeffs[0],
                 intrinsics.coeffs[1],
+                intrinsics.coeffs[4],
                 intrinsics.coeffs[2],
-                intrinsics.coeffs[3],
-                intrinsics.coeffs[4]);
+                intrinsics.coeffs[3]);
 
             _intrinsics.Add(channelName, projTrans);
 
             return projTrans;
         }
 
-        private RealSense2API.RS2StreamProfile GetProfileFromSensor(string channelName)
+        private StreamProfile GetProfileFromSensor(string channelName)
         {
             string sensorName;
             Point2i refResolution;
 
             switch (channelName)
             {
                 case ChannelNames.Color:
-                    sensorName = RealSense2API.SensorName.COLOR;
+                    sensorName = SensorName.COLOR;
                     refResolution = ColorResolution;
                     break;
                 case ChannelNames.ZImage:
                 case ChannelNames.Left:
                 case ChannelNames.Right:
                 case ChannelNames.Distance:
                 default:
-                    sensorName = RealSense2API.SensorName.STEREO;
+                    sensorName = SensorName.STEREO;
                     refResolution = DepthResolution;
                     break;
             }
 
-            RealSense2API.RS2Sensor sensor = RealSense2API.GetSensor(_pipeline, sensorName);
-            RealSense2API.RS2StreamProfilesList list = RealSense2API.GetStreamProfileList(sensor);
-            int count = RealSense2API.GetStreamProfileListCount(list);
+            StreamProfileList list = GetSensor(sensorName).StreamProfiles;
 
-            for (int i = 0; i < count; i++)
+            foreach(VideoStreamProfile profile in list)
             {
-                RealSense2API.RS2StreamProfile p = RealSense2API.GetStreamProfile(list, i);
-                Point2i resolution = RealSense2API.GetStreamProfileResolution(p);
-                if (resolution == refResolution)
-                {
-                    return p;
-                }
+                Point2i res = new Point2i(profile.Width, profile.Height);
+                if (res == refResolution)
+                    return profile;
             }
 
-            return new RealSense2API.RS2StreamProfile(IntPtr.Zero);
+            // return default
```

Comment:
Is it the first profile always the default? Otherwise this should read `// return first profile`

Reply:
We could also throw an exception at this point. The desired profile with the correct resolution could not be found. I'm not sure if this error case is severe enough to stop and throw an exception or to proceed operation with a slightly different StreamProfile.